blocks/header/header.js

```diff
@@ -212,27 +219,43 @@ export default async function decorate(block) {
     decorateIcons(nav);
 
     // Add search form
-    const searchIcon = navSearch.children[0].children[0].children[0];
+    const searchIcon = navSearch.querySelector('.icon');
+    const searchTarget = navSearch.querySelector('a[href]')?.getAttribute('href') || '/search';
     const searchForm = document.createElement('form');
     searchForm.classList.add('search-form');
-    searchForm.setAttribute('action', '/search');
-    navSearch.replaceChild(searchForm, navSearch.children[0]);
-    const searchElement = [...navSearch.children][0];
-    searchElement.innerHTML = `
-      <input type="text" class="search-input" id="search" value="" placeholder="${window.placeholders?.default?.search || 'Search'}" />
-      <button type="submit" class="search-submit"></button>
-    `;
-    const button = searchElement.children[1];
-    button.appendChild(searchIcon);
-    searchElement.addEventListener('submit', (e) => {
+    searchForm.setAttribute('action', searchTarget);
```

Comment:
I don't see the need for this code rewriting and the new code is less performant.

Reply:
after a few google searches like "innerhtml vs appendchild" or "innerhtml vs createElement" the results are saying that innerHTML is also less performant and even less secure than the alternative

Reply:
@jonatan-lledo-netcentric plus innerhtml needs more XSS protection, I prefer it from readability standpoint but security wins

@AngelaFabregues hardcoding the searchTarget was actually a leftover/bug I am fixing with this commit, the path of the search result page should be configurable by the authors, otherwise it makes no sense to use a link for defining the search bar/button in the header